NavigationTools.py

```
@@ -43,12 +48,29 @@ def from_airport_code(cls, location = None) :
         """
         location: a string (KOKC airport code or address) to convert to a set of coordinates
         """
+        error_var = None
+
+        if location == None :
+            raise TypeError(f"Error: {location} was found, please check inputs again.")
         if not(isinstance(location, str)) :
             raise ValueError(f"Error: location is of the wrong type, expected str and got {type(location)}")
         else :
-            location_geocode = GEOLOCATOR.geocode(location)
-            if location_geocode is None :
-                raise RuntimeError(f"Invalid airport. {location} was not found.")
+            try :
+                location_geocode = GEOLOCATOR.geocode(location)
+            except GeocoderUnavailable as e :
+                error_var = GeocoderUnavailable("Error: Remote connection was not established with the geocoding service")
+            except GeocoderTimedOut as e :
+                error_var = GeocoderQueryError("Error: Connection with the geocoding service timed out")
+            except GeocoderQueryError as e :
+                error_var = GeocoderQueryError("Error: Incorrect input was detected when running the geocoding service")
+            except GeocoderServiceError as e :
+                error_var = GeocoderServiceError(f"Error: Geocode services failed on {location} with error message {e}")
+            except GeopyError as e :
+                error_var = GeopyError(f"Error: Geocode failed on {location} with error message {e}")
```

The exceptions `GeocoderUnavailable`, `GeocoderTimedOut`, and `GeocoderServiceError` seem similar enough we could throw the same error message for all of them, I don't think we have to distinguish between them.

---

Any response to this, @SirFlarflajon ?

---

So while I am condesing them into one error catch, there are some slight differences in how they run. How I had it prior it would start with the most specific/common errors and then go to more vague ones.